lib/nexmo_developer/app/services/translator/file_translator.rb

```diff
@@ -0,0 +1,43 @@
+module Translator
+  class FileTranslator
+    attr_reader :doc_path
+
+    def initialize(doc_path)
+      @doc_path = doc_path
+    end
+
+    def translation_requests
+      doc = load_frontmatter
+
+      I18n.available_locales.map do |locale|
+        Translator::TranslationRequest.new(locale: locale, frequency: frequency_calculator(doc), path: doc_path)
+      end
+    end
+
+    def load_frontmatter
+      @load_frontmatter ||= YAML.safe_load(File.read("#{Rails.configuration.docs_base_path}/_documentation/en/#{doc_path}"))
+    end
+
+    def frequency_calculator(doc)
+      return doc['translation_frequency'] if doc['translation_frequency']
+
+      find_matching_product
+    end
+
+    def find_matching_product
```

Comment:
this method is returning a frequency instead of a product, so I would rename it.
Something like this:

```
def frequency
  return frontmatter['translation_frequency'] if frontmatter['translation_frequency']

  product_translation_frequency
end

def product
  # content of find_matching_product goes here memoized
end

def product_translation_frequency
      raise ArgumentError, "Expected a 'translation_frequency' attribute for this product but none found" if product['translation_frequency'].nil?

      product['translation_frequency']
end
```

Reply:
👍